### ISSUE TITLE
How to solve this problem?

### ISSUE THREAD

**Issue by Fitz13009** [modernmt/modernmt#555]
Hi, 

after some unsuccessful training, when I restart the training again, I keep receivinig error report saying: 

`ERROR Unexpected exception: no checkpoints found in ...`

Even if i reinstall this software, I still cannot train.

I don't understand how exactly the entire program is working, I want to clean everything up and retrain the demo model "en to it", how can I do this? Why I need checkpoint to restart the training? 

By the way, I am also curious how Deep Learning (if it is really being used) is working in Java? I only have some knowledge in Tensorflow and PyTorch in Python.

I will appreciate any help or suggestions, thanks in advance. 

Best,
Ge Li

**Comment by EtienneAb3d** [comment]
Hi,

ModernMT is storing your model in the "engines" subfolder and some temporary data in the "runtime" subfolder. If a previous storing exists, it tries to start from last state, and the one you have seems to be an invalid one (no network state stored). 

If you want to restart from scratch, simply delete the contents of these two folders ("engines", and "runtime").

For your Deep Learning question, the MMT team could bring you more precise answers. As far as I know, in MMT it's not directly from Java: for GPU parts, Java codes are calling Python codes.

Etienne

**Comment by nicolabertoldi** [comment]
Hi @Fitz13009 

Etienne is almost right how to reset an engine.
You have to delete directories ```engine/ENGINE``` and ```runtime/ENGINE```, where `ENGINE` is the name you give to your engine. By default ENGINE is `default`.

To resume a training of an engine, which was not completed, you have to re-run the ``mmt create`` adding the flag ``--resume``.

Regarding the error you reported
```ERROR Unexpected exception: no checkpoints found in ...```
please report which was the set of commands which generates the error so that we can reproduce it, together with the full content of the log file
``runtime/ENGINE/logs/node.log``
and the list of the files in the directory 
```runtime/default/tmp/training/```
and its subdirectories.

Finally, ModernMT exploits pytorch and fairseq for the core computation (training and decoding of neural networks).
Java code is mostly used for interfacing the core part with the final user, for normalisation and tokenisation of text, and similar operations.

**Comment by Fitz13009** [comment]
Hi Etienne and NIcola,

thank you for your replies. I found the problem comes from the train-steps. In the readme, it is mentioned that the demo should be run like:

`./mmt create en it examples/data/train/ --train-steps 10000` 

I wanted to save time, so I set the train-steps to 10 and get the error. When I changed it to 100 or 1000, the bug doesn't show up...

Best,
Ge Li

**Comment by nicolabertoldi** [comment]
Hi @Fitz13009 

from the training.log it is seems that training worked properly
And so, the system should have stored the model in the directory ``engines/default``
Specifically, in this directory there should be (at least) these files:
-  ``models/decoder/en__it/model.pt``
-  ``models/decoder/en__it/model.vcb``
-  ``models/decoder/model.conf``
-  ``engine.xconf``

Is this your case?


If so could you please run 
``mmt start -v 2``

and check if the log file exists
``/home/lige/Softwares/mmt/runtime/default/logs/node.log``
and provide us it for our debugging

**Comment by Fitz13009** [comment]
Hi, 

sorry for the late reply, I have all the four files but when I run the command:
`./mmt start -v 2`
 
I receive the same complaints:
 `failed to start node, check log file for more details: /home/lige/Softwares/mmt/runtime/default/logs/node.log`

and I could not find the file "node.log" in the desired directory...

Best,
Ge Li

**Comment by nicolabertoldi** [comment]
@Fitz13009

**Comment by Fitz13009** [comment]
@nicolabertoldi ?

**Comment by nicolabertoldi** [comment]
@Fitz13009 

Are you sure that you the the installation succeded?

What the output of the command ```./mmt status```?

Which is the content of ``models/decoder/model.conf`` and ``engine.xconf``?

PS: In the future, please open a new ticket if the problem is different.
